extensions/sql-migration/src/wizard/accountsSelectionPage.ts

```diff
@@ -0,0 +1,111 @@
+import * as constants from '../models/strings';
+
+export class AccountsSelectionPage extends MigrationWizardPage {
+	private _azureAccountsDropdown!: azdata.DropDownComponent;
+	private _accountsMap: Map<string, azdata.Account> = new Map();
+
+	constructor(wizard: azdata.window.Wizard, migrationStateModel: MigrationStateModel) {
+		super(wizard, azdata.window.createWizardPage(constants.ACCOUNTS_SELECTION_PAGE_TITLE), migrationStateModel);
+	}
+
+	protected async registerContent(view: azdata.ModelView): Promise<void> {
+		const form = view.modelBuilder.formContainer()
+			.withFormItems(
+				[
+					await this.createAzureAccountsDropdown(view)
+				]
+			);
+		await view.initializeModel(form.component());
+		await this.populateAzureAccountsDropdown();
+	}
+
+	private createAzureAccountsDropdown(view: azdata.ModelView): azdata.FormComponent {
+
+		this._azureAccountsDropdown = view.modelBuilder.dropDown().component();
+
+		this._azureAccountsDropdown.onValueChanged(async (value) => {
+			this.migrationStateModel.azureAccount = this._accountsMap.get((this._azureAccountsDropdown.value as azdata.CategoryValue).name)!;
+		});
+
+		const addAccountButton = view.modelBuilder.button()
+			.withProperties<azdata.ButtonProperties>({
+				label: constants.ACCOUNT_ADD_BUTTON_LABEL,
+				width: '100px'
+			})
+			.component();
+
+		addAccountButton.onDidClick(async (event) => {
+			await vscode.commands.executeCommand('workbench.actions.modal.linkedAccount');
+			await this.populateAzureAccountsDropdown();
+		});
+
+		const flexContainer = view.modelBuilder.flexContainer()
+			.withLayout({
+				flexFlow: 'column'
+			})
+			.withItems([this._azureAccountsDropdown, addAccountButton], { CSSStyles: { 'margin': '10px', } })
+			.component();
+
+		return {
+			title: '',
+			component: flexContainer
+		};
+	}
+
+	private async populateAzureAccountsDropdown(): Promise<void> {
+		this._azureAccountsDropdown.loading = true;
+		let accounts = await azdata.accounts.getAllAccounts();
+
+		if (accounts.length === 0) {
+			this._azureAccountsDropdown.value = constants.ACCOUNT_SELECTION_PAGE_NO_LINKED_ACCOUNTS_ERROR;
+			return;
+		}
+
+		this._azureAccountsDropdown.values = accounts.map((account): azdata.CategoryValue => {
+			let accountCategoryValue = {
+				displayName: account.displayInfo.displayName,
+				name: account.displayInfo.userId
+			};
+			this._accountsMap.set(accountCategoryValue.name, account);
+			return accountCategoryValue;
+		});
+
+		this.migrationStateModel.azureAccount = accounts[0];
+		this._azureAccountsDropdown.loading = false;
+	}
+
+	public async onPageEnter(): Promise<void> {
+		this.wizard.registerNavigationValidator((pageChangeInfo: azdata.window.WizardPageChangeInfo) => {
+			if (pageChangeInfo.newPage < pageChangeInfo.lastPage) {
+				this.wizard.message = { text: '' };
+				return true;
+			}
+
+			if (this._azureAccountsDropdown.value === constants.ACCOUNT_SELECTION_PAGE_NO_LINKED_ACCOUNTS_ERROR) {
```

Comment:
You should be able to use the `withValidation` on the dropdown to make it so that the input itself does the validation. This is a lot better since it won't even let users click the next button until they select a valid account.